### ISSUE TITLE
Unneccesary second compilation

### ISSUE THREAD

**Issue by sheavner** [playframework/playframework#4234]
Belated updates for #3954 

The additional compiles seem to be for inner classes.   I've got things in a state where only two classes are affected by the additional compiles: they compile when I execute the command line run command and again when I hit the first page.   If I ^C and start again, they compile immediately after the run command and again when the first page is accessed.    I've shut my IDE.  It's IntelliJ which is usually open, but I don't think it's the problem as the compiles do behave correctly most of the time.

play 2.3.7 | sbt 0.13.7 | scala 2.10.4

```
[debug] All newly invalidated sources after taking into account (previously) recompiled sources:Set()
[debug] Invalidating /.../target/scala-2.10/classes/models/PrimaryModel$InnerClass1.class: stamp changed from lastModified(1429057490000) to lastModified(1429057516000)
[debug] Invalidating /.../target/scala-2.10/classes/models/PrimaryModel$InnerClass2.class: stamp changed from lastModified(1429057490000) to lastModified(1429057516000)

[debug] All initially invalidated sources: Set(/app/services/Services3.java, /app/models/PrimaryModel.java)
[info] Compiling 2 Java sources to /target/scala-2.10/classes...
```

Earlier runs in this session show different mismatched timestamps.

```
[debug] Invalidating /.../target/scala-2.10/classes/models/PrimaryModel$InnerClass1.class: stamp changed from lastModified(1429057309000) to lastModified(1429057315000)
[debug] Invalidating /.../target/scala-2.10/classes/models/PrimaryModel$InnerClass2.class: stamp changed from lastModified(1429057309000) to lastModified(1429057315000)
```

**Comment by marcospereira** [comment]
@sheavner could you try with Play 2.4.0-RC1 and sbt 0.13.8 and report if it is still happening?

**Comment by sheavner** [comment]
I've tried replicating on 2.3.7 and cannot reproduce on demand.   I tried various touch commands on the java source and compiler output $inner.class files, but only get one compile from ./activator compile && ./activator compile.

I've never been able to reproduce it on demand, it just happens sometimes.  But once it does happen, it persists until I clean out /target.   Swapping to another play and sbt build version will probably trigger a clean of /target, but I'll try it if I see the problem in the next few weeks.
